flake8_trio.py

```diff
@@ -55,6 +55,10 @@
         "`trio.[fail/move_on]_[after/at]` instead"
     ),
     "TRIO110": "`while <condition>: await trio.sleep()` should be replaced by a `trio.Event`.",
+    "TRIO111": (
+        "variable {2}, from context manager on line {0}, "
+        "passed to {3} from nursery opened on {1}, might get closed while in use"
```

Comment:
```suggestion
        "variable {} is usable within the context manager on line {}, but that "
        "will close before nursery opened on line {} - this is usually a bug.  "
        "Nurseries should generally be the inner-most context manager."
```

Not *super* happy with this, but I think it gets at the core problem pretty directly.  (message also requires some changes to the construction of the error, but dropping one and rearranging isn't too hard.)

Reply:
No need to drop/rearrange with indexed args, and redundant args getting ignored :P Long term maybe nice to do, but I think I'll leave the redundant arg for now for ease of putting it back.

There's also a question on how wordy one wants the error message to be, I personally prefer messages to be short enough to get shown as a one-liner in a status bar - esp the gist of them.

I'd maybe do something like
> variable {2}, from context manager on line {0}, passed to {3} from nursery opened outside of it on line {1}, might get invalidly accessed due to context manager closing before the nursery.
(though not sure about invalidly vs illegally vs ??)
But I'll leave yours as is and let you decide.